mp/src/game/client/momentum/ui/HUD/hud_stickybombs.cpp

```diff
@@ -67,19 +76,54 @@ void CHudStickybombs::OnThink()
         return;
 
     const auto pStickylauncher = static_cast<CMomentumStickybombLauncher *>(pStickyPtr);
-
     const auto iStickybombs = pStickylauncher->GetStickybombCount();
-    if (mom_hud_sj_stickycount_autohide.GetBool())
+
+    for (int i = 0; i < MOM_WEAPON_STICKYBOMB_COUNT; i++)
     {
-        m_pStickyLabel->SetVisible(iStickybombs > 0);
+        if (i < iStickybombs)
+        {
+            bool bCanExplode = true;
+            const auto pSticky = pStickylauncher->GetStickyByCount(i);
+            if (pSticky)
+            {
+                bCanExplode = pSticky->CanExplode();
+            }
+            
+            m_Stickyboxes[i]->Process(bCanExplode ? STICKYBOX_PROCTYPE_ACTIVE : STICKYBOX_PROCTYPE_DISABLED);
+        }
+        else
+        {
+            m_Stickyboxes[i]->Process(STICKYBOX_PROCTYPE_NOSTICKY);
```

Comment:
I'm not sure I understand why process takes an enum. They're all pretty distinct calls with no overlapping, so why not just call the individual process functions here?